src/dbt_client/dbtTerminal.ts

```diff
@@ -15,12 +26,57 @@ export class DBTTerminal {
   }
 
   log(message: string) {
-    this.outputChannel.append(stripANSI(message));
+    this.outputChannel.info(stripANSI(message));
     if (this.terminal !== undefined) {
       this.writeEmitter.fire(message);
     }
   }
 
+  trace(msg: string) {
+    this.outputChannel?.appendLine(stripANSI(msg));
+    console.log(msg);
+  }
+
+  debug(msg: string) {
+    this.outputChannel?.debug(stripANSI(msg));
+    console.debug(msg);
+  }
+
+  info(e: CustomException, sendTelemetry: boolean = false) {
+    const errorMessage = e.getErrorMessage();
+    this.outputChannel?.info(stripANSI(errorMessage));
+    console.info(errorMessage);
+    if (sendTelemetry) {
+      this.telemetry.sendTelemetryEvent(e.name, e.extra);
+    }
+  }
+
+  warn(e: CustomException, sendTelemetry: boolean = false) {
+    const errorMessage = e.getErrorMessage();
+    this.outputChannel?.warn(stripANSI(errorMessage));
+    console.warn(errorMessage);
+    if (sendTelemetry) {
+      this.telemetry.sendTelemetryError(e.name, e.error, e.extra);
+    }
+  }
+
+  error(e: CustomException, sendTelemetry = false) {
+    const errorMessage = e.getErrorMessage();
+    this.outputChannel?.error(stripANSI(errorMessage));
+    window.showErrorMessage(extendErrorWithSupportLinks(e.message));
```

Comment:
Showing error messages goes beyond what a terminal is supposed to do. Maybe think of a better name. Why not keep this part where it was?